### PR TITLE
AttentionOCR partial TF2 migration

### PR DESCRIPTION
**Pull request by kyscg**
# Description

This PR aims to migrate the TensorFlow code from 1.x to 2.x for the AttentionOCR model in the `research` folder.

The changes will partly address #8703 but there is more work to do.

Further work will include
* Consolidating documentation
* Providing tutorials

## Type of change

- [x] Documentation update
- [x] TensorFlow 2 migration

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [ ] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.

### CODE REVIEW

**Comment by kyscg** [comment]
I think `tf.image` requires a manual check: `align_corners` is not supported by `tf.image.resize`, the new default transformation is close to what v1 provided. Maybe `compat.v1.image.resize_bilinear` will do the job

File: research/attention_ocr/python/model.py [hunk 1]
```diff
@@ -665,13 +669,14 @@ def sname(label):
     # tf.summary.text(sname('text/pr'), pr_text)
     # gt_text = charset_mapper.get_text(data.labels[:max_outputs,:])
     # tf.summary.text(sname('text/gt'), gt_text)
```
Comment (xavigibert) [inline research/attention_ocr/python/model.py:671]: Can you uncomment this lines and, if necessary, replace tf.summary.text with tf.compat.v1.summary.text?
You can do it as a follow-up PR.

**Comment by xavigibert** [comment]
I suggest that you rename this PR as "AttentionOCR partial TF2 migration", since the code is still not TF2 compatible because it depends on  tensorflow.contrib.

File: research/attention_ocr/python/model.py (hunk 1 above)
Comment (kyscg) [reply research/attention_ocr/python/model.py:671]: Yes, I'll do that in another PR

**Pull request merged**